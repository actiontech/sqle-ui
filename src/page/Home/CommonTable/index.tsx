import { Badge } from 'antd';
import { IWorkflowDetailResV2 } from '../../../api/common';
import { TableColumn } from '../../../types/common.type';
import CommonTable from './Table';

export const genTabPaneTitle = (title: string, badgeCount?: number) => {
  return badgeCount ? (
    <Badge
      className="tabs-panel-badge"
      count={badgeCount}
      overflowCount={99}
      size="small"
    >
      <span className="tabs-panel-title">{title}</span>
    </Badge>
  ) : (
    title
  );
};

export type CommonTableInfoType = {
  data: IWorkflowDetailResV2[];
  error: Error | undefined;
  loading: boolean;
};

export interface ICommonTableProps {
  tableInfo: CommonTableInfoType;
  customColumn?: () => TableColumn<IWorkflowDetailResV2>;
}

export const DASHBOARD_COMMON_GET_ORDER_NUMBER = 5;

export default CommonTable;
