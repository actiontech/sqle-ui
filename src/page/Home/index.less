.sqle-home-content {
  .tabs-panel-badge {
    color: inherit;
  }
  .tabs-panel-title {
    margin: 10px 18px 0 0;
  }
  .ant-card-head-title {
    padding: 16px 0 6px;
  }
}
