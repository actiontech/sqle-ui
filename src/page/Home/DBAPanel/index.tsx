import { SyncOutlined } from '@ant-design/icons';
import { Card, Space, Tabs, TabsProps, Typography } from 'antd';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { getWorkflowsV1FilterStatusEnum } from '../../../api/workflow/index.enum';
import { IReduxState } from '../../../store';
import CommonTable, {
  DASHBOARD_COMMON_GET_ORDER_NUMBER,
  genTabPaneTitle,
} from '../CommonTable';
import useDashboardRequest from '../useDashboardRequest';
import { IDBAPanelProps } from './index.type';

enum tabsKeyEnum {
  needMeReview = '1',
  needMeExec = '2',
}

const DBAPanel: React.FC<IDBAPanelProps> = ({
  workflowStatistics,
  getWorkflowStatistics,
  projectName,
}) => {
  const username = useSelector<IReduxState, string>(
    (state) => state.user.username
  );
  const [currentActiveKey, setCurrentActiveKey] = useState<tabsKeyEnum>(
    tabsKeyEnum.needMeReview
  );
  const { t } = useTranslation();

  const handleChangeTabs: TabsProps['onChange'] = (key) => {
    setCurrentActiveKey(key as tabsKeyEnum);
  };

  const needMeReviewResponse = useDashboardRequest(
    {
      page_index: 1,
      page_size: DASHBOARD_COMMON_GET_ORDER_NUMBER,
      filter_current_step_assignee_user_name: username,
      filter_status: getWorkflowsV1FilterStatusEnum.wait_for_audit,
      project_name: projectName,
    },
    [projectName]
  );

  const needMeExecResponse = useDashboardRequest(
    {
      page_index: 1,
      page_size: DASHBOARD_COMMON_GET_ORDER_NUMBER,
      filter_current_step_assignee_user_name: username,
      filter_status: getWorkflowsV1FilterStatusEnum.wait_for_execution,
      project_name: projectName,
    },
    [projectName]
  );

  const tableLoading =
    needMeExecResponse.loading || needMeReviewResponse.loading;

  const refreshTable = () => {
    if (tableLoading) {
      return;
    }

    needMeExecResponse.refresh();
    needMeReviewResponse.refresh();
    getWorkflowStatistics();
  };

  const tabItems: TabsProps['items'] = [
    {
      key: tabsKeyEnum.needMeReview,
      label: genTabPaneTitle(
        t('dashboard.pendingOrder.needMeReview'),
        workflowStatistics?.need_me_to_review_workflow_number
      ),
      children: (
        <CommonTable
          tableInfo={{
            loading: needMeReviewResponse.loading,
            data: needMeReviewResponse.data ?? [],
            error: needMeReviewResponse.error,
          }}
        />
      ),
    },
    {
      key: tabsKeyEnum.needMeExec,
      label: genTabPaneTitle(
        t('dashboard.pendingOrder.needMeExec'),
        workflowStatistics?.need_me_to_execute_workflow_number
      ),
      children: (
        <CommonTable
          tableInfo={{
            loading: needMeExecResponse.loading,
            data: needMeExecResponse.data ?? [],
            error: needMeExecResponse.error,
          }}
        />
      ),
    },
  ];

  return (
    <Card className="full-width-element">
      <Tabs
        animated={true}
        size="small"
        type="card"
        onChange={handleChangeTabs}
        activeKey={currentActiveKey}
        tabBarExtraContent={{
          left: (
            <span className="tab-panel-title">
              {t('dashboard.title.pendingOrder')}
            </span>
          ),
          right: (
            <Space>
              <Typography.Text>
                {t('dashboard.tableLimitTips', {
                  number: DASHBOARD_COMMON_GET_ORDER_NUMBER,
                })}
              </Typography.Text>
              <SyncOutlined
                data-testid="refreshTable"
                spin={tableLoading}
                onClick={refreshTable}
              />
            </Space>
          ),
        }}
        items={tabItems}
      />
    </Card>
  );
};

export default DBAPanel;
