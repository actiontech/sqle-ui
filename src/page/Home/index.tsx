import { useRequest } from 'ahooks';
import { PageHeader, Space } from 'antd';
import { useTranslation } from 'react-i18next';
import dashboard from '../../api/dashboard';
import DBAPanel from './DBAPanel';
import QuickLink from './QuickLink';
import DEVPanel from './DEVPanel';
import RecentlyOrderPanel from './RecentlyOrderPanel';

import './index.less';

const Home = () => {
  const { t } = useTranslation();

  const { data: workflowStatistics } = useRequest(
    () => dashboard.getDashboardV1(),
    {
      formatResult(res) {
        return res.data?.data?.workflow_statistics;
      },
    }
  );

  return (
    <>
      <PageHeader title={t('dashboard.pageTitle')} ghost={false} />
      <section className="padding-content sqle-home-content">
        <Space direction="vertical" className="full-width-element" size="large">
          <DBAPanel workflowStatistics={workflowStatistics} />
          <DEVPanel workflowStatistics={workflowStatistics} />
          <RecentlyOrderPanel />
        </Space>
      </section>
      <QuickLink />
    </>
  );
};

export default Home;
