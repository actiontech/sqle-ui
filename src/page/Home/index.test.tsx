import { cleanup, waitFor } from '@testing-library/react';
import Home from '.';
import dashboard from '../../api/dashboard';
import workflow from '../../api/workflow';
import { renderWithRouter } from '../../testUtils/customRender';
import { mockUseSelector } from '../../testUtils/mockRedux';
import { resolveThreeSecond } from '../../testUtils/mockRequest';

describe('Home', () => {
  const mockGetDashboardV1 = () => {
    const spy = jest.spyOn(dashboard, 'getDashboardV1');
    spy.mockImplementation(() =>
      resolveThreeSecond({
        workflow_statistics: {
          my_need_review_workflow_number: 2,

          my_need_execute_workflow_number: 99,

          my_rejected_workflow_number: 0,

          need_me_to_execute_workflow_number: 80,

          need_me_to_review_workflow_number: 102,
        },
      })
    );
    return spy;
  };
  const mockRequest = () => {
    const spy = jest.spyOn(workflow, 'getGlobalWorkflowsV1');
    spy.mockImplementation(() => resolveThreeSecond([]));
    return spy;
  };

  beforeEach(() => {
    jest.useFakeTimers();
    mockUseSelector({ user: { username: 'admin' } });
    mockRequest();
  });
  afterEach(() => {
    jest.clearAllMocks();
    jest.useRealTimers();

    cleanup();
  });

  test('should match snapshot', async () => {
    mockGetDashboardV1();
    const { container } = renderWithRouter(<Home />);

    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });

    // fix local snapshot are different from github actions
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });

    expect(container).toMatchSnapshot();
  });

  test('should be called getDashboardV1 interface', async () => {
    const getDashboardSpy = mockGetDashboardV1();
    expect(getDashboardSpy).toBeCalledTimes(0);
    renderWithRouter(<Home />);

    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });

    expect(getDashboardSpy).toBeCalledTimes(1);
  });
});
