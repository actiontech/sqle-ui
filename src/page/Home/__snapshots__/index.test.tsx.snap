// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div>
  <div
    class="ant-page-header"
  >
    <div
      class="ant-page-header-heading"
    >
      <div
        class="ant-page-header-heading-left"
      >
        <span
          class="ant-page-header-heading-title"
          title="dashboard.pageTitle"
        >
          dashboard.pageTitle
        </span>
      </div>
    </div>
  </div>
  <section
    class="padding-content sqle-home-content"
  >
    <div
      class="ant-space ant-space-vertical full-width-element"
    >
      <div
        class="ant-space-item"
        style="margin-bottom: 24px;"
      >
        <div
          class="ant-card ant-card-bordered full-width-element"
        >
          <div
            class="ant-card-body"
          >
            <div
              class="ant-tabs ant-tabs-top ant-tabs-small ant-tabs-card"
            >
              <div
                class="ant-tabs-nav"
                role="tablist"
              >
                <div
                  class="ant-tabs-nav-wrap"
                >
                  <div
                    class="ant-tabs-nav-list"
                    style="transform: translate(0px, 0px);"
                  >
                    <div
                      class="ant-tabs-tab ant-tabs-tab-active"
                    >
                      <div
                        aria-controls="rc-tabs-test-panel-1"
                        aria-selected="true"
                        class="ant-tabs-tab-btn"
                        id="rc-tabs-test-tab-1"
                        role="tab"
                        tabindex="0"
                      >
                        <span
                          class="ant-badge tabs-panel-badge"
                        >
                          <span
                            class="tabs-panel-title"
                          >
                            dashboard.title.needMeReview
                          </span>
                          <sup
                            class="ant-scroll-number ant-badge-count ant-badge-count-sm ant-badge-multiple-words"
                            data-show="true"
                            title="102"
                          >
                            99+
                          </sup>
                        </span>
                      </div>
                    </div>
                    <div
                      class="ant-tabs-tab"
                    >
                      <div
                        aria-controls="rc-tabs-test-panel-2"
                        aria-selected="false"
                        class="ant-tabs-tab-btn"
                        id="rc-tabs-test-tab-2"
                        role="tab"
                        tabindex="0"
                      >
                        <span
                          class="ant-badge tabs-panel-badge"
                        >
                          <span
                            class="tabs-panel-title"
                          >
                            dashboard.title.needMeExec
                          </span>
                          <sup
                            class="ant-scroll-number ant-badge-count ant-badge-count-sm ant-badge-multiple-words"
                            data-show="true"
                            title="80"
                          >
                            <span
                              class="ant-scroll-number-only"
                              style="transition: none;"
                            >
                              <span
                                class="ant-scroll-number-only-unit current"
                              >
                                8
                              </span>
                            </span>
                            <span
                              class="ant-scroll-number-only"
                              style="transition: none;"
                            >
                              <span
                                class="ant-scroll-number-only-unit current"
                              >
                                0
                              </span>
                            </span>
                          </sup>
                        </span>
                      </div>
                    </div>
                    <div
                      class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                    />
                  </div>
                </div>
                <div
                  class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
                >
                  <button
                    aria-controls="rc-tabs-test-more-popup"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-hidden="true"
                    class="ant-tabs-nav-more"
                    id="rc-tabs-test-more"
                    style="visibility: hidden; order: 1;"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="ellipsis"
                      class="anticon anticon-ellipsis"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="ellipsis"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                <div
                  class="ant-tabs-extra-content"
                >
                  <span
                    aria-label="sync"
                    class="anticon anticon-sync"
                    data-testid="refreshTable"
                    role="img"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="sync"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 01755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 01512.1 856a342.24 342.24 0 01-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 00-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 00-8-8.2z"
                      />
                    </svg>
                  </span>
                  <button
                    class="ant-btn ant-btn-link"
                    type="button"
                  >
                    <span>
                      common.showAll
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="ant-tabs-content-holder"
              >
                <div
                  class="ant-tabs-content ant-tabs-content-top"
                >
                  <div
                    aria-hidden="false"
                    aria-labelledby="rc-tabs-test-tab-1"
                    class="ant-tabs-tabpane ant-tabs-tabpane-active"
                    id="rc-tabs-test-panel-1"
                    role="tabpanel"
                    tabindex="0"
                  >
                    <div
                      class="ant-table-wrapper"
                    >
                      <div
                        class="ant-spin-nested-loading"
                      >
                        <div
                          class="ant-spin-container"
                        >
                          <div
                            class="ant-table ant-table-empty"
                          >
                            <div
                              class="ant-table-container"
                            >
                              <div
                                class="ant-table-content"
                              >
                                <table
                                  style="table-layout: auto;"
                                >
                                  <colgroup />
                                  <thead
                                    class="ant-table-thead"
                                  >
                                    <tr>
                                      <th
                                        class="ant-table-cell"
                                      >
                                        order.order.name
                                      </th>
                                      <th
                                        class="ant-table-cell"
                                      >
                                        order.order.desc
                                      </th>
                                      <th
                                        class="ant-table-cell"
                                      >
                                        order.order.createTime
                                      </th>
                                    </tr>
                                  </thead>
                                  <tbody
                                    class="ant-table-tbody"
                                  >
                                    <tr
                                      class="ant-table-placeholder"
                                    >
                                      <td
                                        class="ant-table-cell"
                                        colspan="3"
                                      >
                                        <div
                                          class="ant-empty ant-empty-normal"
                                        >
                                          <div
                                            class="ant-empty-image"
                                          >
                                            <svg
                                              class="ant-empty-img-simple"
                                              height="41"
                                              viewBox="0 0 64 41"
                                              width="64"
                                              xmlns="http://www.w3.org/2000/svg"
                                            >
                                              <g
                                                fill="none"
                                                fill-rule="evenodd"
                                                transform="translate(0 1)"
                                              >
                                                <ellipse
                                                  class="ant-empty-img-simple-ellipse"
                                                  cx="32"
                                                  cy="33"
                                                  rx="32"
                                                  ry="7"
                                                />
                                                <g
                                                  class="ant-empty-img-simple-g"
                                                  fill-rule="nonzero"
                                                >
                                                  <path
                                                    d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                                  />
                                                  <path
                                                    class="ant-empty-img-simple-path"
                                                    d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                                  />
                                                </g>
                                              </g>
                                            </svg>
                                          </div>
                                          <div
                                            class="ant-empty-description"
                                          >
                                            No Data
                                          </div>
                                        </div>
                                      </td>
                                    </tr>
                                  </tbody>
                                </table>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    aria-labelledby="rc-tabs-test-tab-2"
                    class="ant-tabs-tabpane"
                    id="rc-tabs-test-panel-2"
                    role="tabpanel"
                    style="display: none;"
                    tabindex="-1"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-space-item"
        style="margin-bottom: 24px;"
      >
        <div
          class="ant-card ant-card-bordered full-width-element"
        >
          <div
            class="ant-card-body"
          >
            <div
              class="ant-tabs ant-tabs-top ant-tabs-small ant-tabs-card"
            >
              <div
                class="ant-tabs-nav"
                role="tablist"
              >
                <div
                  class="ant-tabs-nav-wrap"
                >
                  <div
                    class="ant-tabs-nav-list"
                    style="transform: translate(0px, 0px);"
                  >
                    <div
                      class="ant-tabs-tab ant-tabs-tab-active"
                    >
                      <div
                        aria-controls="rc-tabs-test-panel-1"
                        aria-selected="true"
                        class="ant-tabs-tab-btn"
                        id="rc-tabs-test-tab-1"
                        role="tab"
                        tabindex="0"
                      >
                        <span
                          class="ant-badge tabs-panel-badge"
                        >
                          <span
                            class="tabs-panel-title"
                          >
                            dashboard.title.pendingReviewByMe
                          </span>
                          <sup
                            class="ant-scroll-number ant-badge-count ant-badge-count-sm"
                            data-show="true"
                            title="2"
                          >
                            <span
                              class="ant-scroll-number-only"
                              style="transition: none;"
                            >
                              <span
                                class="ant-scroll-number-only-unit current"
                              >
                                2
                              </span>
                            </span>
                          </sup>
                        </span>
                      </div>
                    </div>
                    <div
                      class="ant-tabs-tab"
                    >
                      <div
                        aria-controls="rc-tabs-test-panel-2"
                        aria-selected="false"
                        class="ant-tabs-tab-btn"
                        id="rc-tabs-test-tab-2"
                        role="tab"
                        tabindex="0"
                      >
                        <span
                          class="ant-badge tabs-panel-badge"
                        >
                          <span
                            class="tabs-panel-title"
                          >
                            dashboard.title.pendingExecByMe
                          </span>
                          <sup
                            class="ant-scroll-number ant-badge-count ant-badge-count-sm ant-badge-multiple-words"
                            data-show="true"
                            title="99"
                          >
                            <span
                              class="ant-scroll-number-only"
                              style="transition: none;"
                            >
                              <span
                                class="ant-scroll-number-only-unit current"
                              >
                                9
                              </span>
                            </span>
                            <span
                              class="ant-scroll-number-only"
                              style="transition: none;"
                            >
                              <span
                                class="ant-scroll-number-only-unit current"
                              >
                                9
                              </span>
                            </span>
                          </sup>
                        </span>
                      </div>
                    </div>
                    <div
                      class="ant-tabs-tab"
                    >
                      <div
                        aria-controls="rc-tabs-test-panel-3"
                        aria-selected="false"
                        class="ant-tabs-tab-btn"
                        id="rc-tabs-test-tab-3"
                        role="tab"
                        tabindex="0"
                      >
                        dashboard.title.rejectedOrderByMe
                      </div>
                    </div>
                    <div
                      class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                    />
                  </div>
                </div>
                <div
                  class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
                >
                  <button
                    aria-controls="rc-tabs-test-more-popup"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-hidden="true"
                    class="ant-tabs-nav-more"
                    id="rc-tabs-test-more"
                    style="visibility: hidden; order: 1;"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="ellipsis"
                      class="anticon anticon-ellipsis"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="ellipsis"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                <div
                  class="ant-tabs-extra-content"
                >
                  <span
                    aria-label="sync"
                    class="anticon anticon-sync"
                    data-testid="refreshTable"
                    role="img"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="sync"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 01755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 01512.1 856a342.24 342.24 0 01-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 00-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 00-8-8.2z"
                      />
                    </svg>
                  </span>
                  <button
                    class="ant-btn ant-btn-link"
                    type="button"
                  >
                    <span>
                      common.showAll
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="ant-tabs-content-holder"
              >
                <div
                  class="ant-tabs-content ant-tabs-content-top"
                >
                  <div
                    aria-hidden="false"
                    aria-labelledby="rc-tabs-test-tab-1"
                    class="ant-tabs-tabpane ant-tabs-tabpane-active"
                    id="rc-tabs-test-panel-1"
                    role="tabpanel"
                    tabindex="0"
                  >
                    <div
                      class="ant-table-wrapper"
                    >
                      <div
                        class="ant-spin-nested-loading"
                      >
                        <div
                          class="ant-spin-container"
                        >
                          <div
                            class="ant-table ant-table-empty"
                          >
                            <div
                              class="ant-table-container"
                            >
                              <div
                                class="ant-table-content"
                              >
                                <table
                                  style="table-layout: auto;"
                                >
                                  <colgroup />
                                  <thead
                                    class="ant-table-thead"
                                  >
                                    <tr>
                                      <th
                                        class="ant-table-cell"
                                      >
                                        order.order.name
                                      </th>
                                      <th
                                        class="ant-table-cell"
                                      >
                                        order.order.desc
                                      </th>
                                      <th
                                        class="ant-table-cell"
                                      >
                                        order.order.createTime
                                      </th>
                                    </tr>
                                  </thead>
                                  <tbody
                                    class="ant-table-tbody"
                                  >
                                    <tr
                                      class="ant-table-placeholder"
                                    >
                                      <td
                                        class="ant-table-cell"
                                        colspan="3"
                                      >
                                        <div
                                          class="ant-empty ant-empty-normal"
                                        >
                                          <div
                                            class="ant-empty-image"
                                          >
                                            <svg
                                              class="ant-empty-img-simple"
                                              height="41"
                                              viewBox="0 0 64 41"
                                              width="64"
                                              xmlns="http://www.w3.org/2000/svg"
                                            >
                                              <g
                                                fill="none"
                                                fill-rule="evenodd"
                                                transform="translate(0 1)"
                                              >
                                                <ellipse
                                                  class="ant-empty-img-simple-ellipse"
                                                  cx="32"
                                                  cy="33"
                                                  rx="32"
                                                  ry="7"
                                                />
                                                <g
                                                  class="ant-empty-img-simple-g"
                                                  fill-rule="nonzero"
                                                >
                                                  <path
                                                    d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                                  />
                                                  <path
                                                    class="ant-empty-img-simple-path"
                                                    d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                                  />
                                                </g>
                                              </g>
                                            </svg>
                                          </div>
                                          <div
                                            class="ant-empty-description"
                                          >
                                            No Data
                                          </div>
                                        </div>
                                      </td>
                                    </tr>
                                  </tbody>
                                </table>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    aria-labelledby="rc-tabs-test-tab-2"
                    class="ant-tabs-tabpane"
                    id="rc-tabs-test-panel-2"
                    role="tabpanel"
                    style="display: none;"
                    tabindex="-1"
                  />
                  <div
                    aria-hidden="true"
                    aria-labelledby="rc-tabs-test-tab-3"
                    class="ant-tabs-tabpane"
                    id="rc-tabs-test-panel-3"
                    role="tabpanel"
                    style="display: none;"
                    tabindex="-1"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          class="ant-card ant-card-bordered"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center flex-space-between"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <span>
                      dashboard.title.recentlyOnlineWorkOrder
                    </span>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <div>
                      <span
                        aria-label="sync"
                        class="anticon anticon-sync"
                        data-testid="refreshTable"
                        role="img"
                        style="font-size: 14px;"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="sync"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 01755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 01512.1 856a342.24 342.24 0 01-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 00-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 00-8-8.2z"
                          />
                        </svg>
                      </span>
                      <button
                        class="ant-btn ant-btn-link"
                        type="button"
                      >
                        <span>
                          common.showAll
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="ant-table-wrapper"
            >
              <div
                class="ant-spin-nested-loading"
              >
                <div
                  class="ant-spin-container"
                >
                  <div
                    class="ant-table ant-table-empty"
                  >
                    <div
                      class="ant-table-container"
                    >
                      <div
                        class="ant-table-content"
                      >
                        <table
                          style="table-layout: auto;"
                        >
                          <colgroup />
                          <thead
                            class="ant-table-thead"
                          >
                            <tr>
                              <th
                                class="ant-table-cell"
                              >
                                order.order.name
                              </th>
                              <th
                                class="ant-table-cell"
                              >
                                order.order.desc
                              </th>
                              <th
                                class="ant-table-cell"
                              >
                                order.order.createTime
                              </th>
                              <th
                                class="ant-table-cell"
                              >
                                order.order.status
                              </th>
                            </tr>
                          </thead>
                          <tbody
                            class="ant-table-tbody"
                          >
                            <tr
                              class="ant-table-placeholder"
                            >
                              <td
                                class="ant-table-cell"
                                colspan="4"
                              >
                                <div
                                  class="ant-empty ant-empty-normal"
                                >
                                  <div
                                    class="ant-empty-image"
                                  >
                                    <svg
                                      class="ant-empty-img-simple"
                                      height="41"
                                      viewBox="0 0 64 41"
                                      width="64"
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <g
                                        fill="none"
                                        fill-rule="evenodd"
                                        transform="translate(0 1)"
                                      >
                                        <ellipse
                                          class="ant-empty-img-simple-ellipse"
                                          cx="32"
                                          cy="33"
                                          rx="32"
                                          ry="7"
                                        />
                                        <g
                                          class="ant-empty-img-simple-g"
                                          fill-rule="nonzero"
                                        >
                                          <path
                                            d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                          />
                                          <path
                                            class="ant-empty-img-simple-path"
                                            d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                          />
                                        </g>
                                      </g>
                                    </svg>
                                  </div>
                                  <div
                                    class="ant-empty-description"
                                  >
                                    No Data
                                  </div>
                                </div>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <button
    class="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg fixed-widgets-dashboard-namespace"
    type="button"
  >
    <span
      aria-label="plus"
      class="anticon anticon-plus"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="plus"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <defs>
          <style />
        </defs>
        <path
          d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
        />
        <path
          d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
        />
      </svg>
    </span>
  </button>
</div>
`;
